examples/simon_algorithm.py

```diff
@@ -0,0 +1,161 @@
+import random
+# Used for classical post-processing:
+from collections import Counter
+import numpy as np
+import scipy as sp
+import cirq
+"""Demonstrates Simon's algorithm.
+Simon's Algorithm solves the following problem: 
+
+Given a function  f:{0,1}^n -> {0,1}^n, such that for some s ∈ {0,1}^n,  
+
+f(x) = f(y) iff  x+y ∈ {0^n, s}, 
+
+find the n-bit string s. 
+
+A classical algorithm requires O(2^n/2) queries to find s, while Simon’s algorithm needs only O(n) quantum queries.  
+=== REFERENCE ===
+D. R. Simon. On the power of quantum cryptography. In35th FOCS, pages 116–123, Santa Fe,New Mexico, 1994. IEEE Computer Society Press.
+
+=== EXAMPLE OUTPUT ===
+Secret string = [1, 0, 0, 1, 0, 0]
+Circuit:
+                ┌──────┐   ┌───────────┐
+(0, 0): ────H────@──────────@─────@──────H───M('result')───
+                 │          │     │          │
+(1, 0): ────H────┼@─────────┼─────┼──────H───M─────────────
+                 ││         │     │          │
+(2, 0): ────H────┼┼@────────┼─────┼──────H───M─────────────
+                 │││        │     │          │
+(3, 0): ────H────┼┼┼@───────┼─────┼──────H───M─────────────
+                 ││││       │     │          │
+(4, 0): ────H────┼┼┼┼@──────┼─────┼──────H───M─────────────
+                 │││││      │     │          │
+(5, 0): ────H────┼┼┼┼┼@─────┼─────┼──────H───M─────────────
+                 ││││││     │     │
+(6, 0): ─────────X┼┼┼┼┼─────X─────┼───×────────────────────
+                  │││││           │   │
+(7, 0): ──────────X┼┼┼┼───────────┼───┼────────────────────
+                   ││││           │   │
+(8, 0): ───────────X┼┼┼───────────┼───┼────────────────────
+                    │││           │   │
+(9, 0): ────────────X┼┼───────────X───×────────────────────
+                     ││
+(10, 0): ────────────X┼────────────────────────────────────
+                      │
+(11, 0): ─────────────X────────────────────────────────────
+                └──────┘   └───────────┘
+Most common Simon Algorithm answer is: ('[1 0 0 1 0 0]', 100)
+
+***If the input string is s=0^n, no significant answer can be distinguished (since the
+null-space of the system of equations provided by the measurements gives a random vector). 
+In this case the output is: 
+"No significant answers obtained. Secret Sequence is probably [0,0,0,0,0,0]"
+"""
+
+
+def main(qubit_count=6):
```

Comment:
Reducing the qubit_count to 3 will still get the point across while saving some time on our tests